src/utils/clock.cc

```diff
@@ -0,0 +1,61 @@
+/************************************************************
+*
+* Licensed to the Apache Software Foundation (ASF) under one
+* or more contributor license agreements.  See the NOTICE file
+* distributed with this work for additional information
+* regarding copyright ownership.  The ASF licenses this file
+* to you under the Apache License, Version 2.0 (the
+* "License"); you may not use this file except in compliance
+* with the License.  You may obtain a copy of the License at
+*
+*   http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied.  See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*
+*************************************************************/
+
+#include "singa/utils/clock.h"
+
+#include <glog/logging.h>
+#include <cmath>
+#include <ctime>
+#include "singa/utils/cluster.h"
+#include "singa/utils/factory.h"
+#include "singa/utils/singleton.h"
+#include "singa/utils/context.h"
+
+namespace singa {
+
+void Clock::start() {
+	start_t = clock();
+}
+
+double Clock::end() {
+	end_t=clock();
+	double runtime = (end_t-start_t)/double(CLOCKS_PER_SEC)*1000;
+	return runtime;
+}
+
+double Clock::end(int iteration_num, std::string content) {
+ end_t=clock();
+ double runtime=(end_t-start_t)/(double(CLOCKS_PER_SEC)*1000*iteration_num);
+ return runtime;
+}
+
+void Clock::endWithLog( std::string content) {
```

Comment:
The function names should start with upper case letters except set/get functions.
Would it be better to use EndWithDLog as you used DLog for logging?